client/src/components/SubmissionTable.vue

```diff
@@ -0,0 +1,162 @@
+<template>
+  <q-table
+    bordered
+    flat
+    square
+    :columns="cols"
+    :rows="tableData"
+    row-key="id"
+    :filter="status_filter"
+    dense
+  >
+    <template #top>
+      <div class="row full-width justify-between q-pb-md">
+        <div class="column">
+          <h3 class="q-my-none">{{ $t(title) }}</h3>
+          <!-- eslint-disable-next-line vue/no-v-html -->
+          <p class="q-mb-none" v-html="$t(byline)"></p>
+        </div>
+        <div class="column">
+          <q-select
+            v-model="status_filter"
+            clearable
+            square
+            outlined
+            dense
+            :label="$t(`submission_tables.filter_label`)"
+            :options="unique_statuses"
+            style="width: 240px"
+            class="q-mt-md q-mr-xs"
+          >
+            <template #selected-item="scope">
+              {{ $t(`submission.status.${scope.opt}`) }}
+            </template>
+            <template #option="scope">
+              <q-item v-bind="scope.itemProps">
+                <q-item-section>
+                  <q-item-label>
+                    {{ $t(`submission.status.${scope.opt}`) }}
+                  </q-item-label>
+                </q-item-section>
+              </q-item>
+            </template>
+          </q-select>
+        </div>
+      </div>
+    </template>
+    <template #no-data>
+      <div class="full-width row flex-center text--grey q-py-lg">
+        <p class="text-h3">{{ $t(noData) }}</p>
+      </div>
+    </template>
+    <template #body="props">
+      <q-tr :props="props">
+        <q-td key="id" :props="props">
+          {{ props.row.id }}
+        </q-td>
+        <q-td key="title" :props="props">
+          <router-link
+            :to="{
+              name: 'submission_review',
+              params: { id: props.row.id },
+            }"
+            >{{ props.row.title }}
+          </router-link>
+        </q-td>
+        <q-td key="publication" :props="props">
+          <router-link
+            :to="{
+              name: 'publication:home',
+              params: { id: props.row.publication.id },
+            }"
+            >{{ props.row.publication.name }}
+          </router-link>
+        </q-td>
+        <q-td key="status" :props="props">
+          {{ $t(`submission.status.${props.row.status}`) }}
+        </q-td>
+        <q-td key="actions" :props="props">
+          <submission-table-actions
+            :submission="props.row"
+            action-type="reviews"
+            flat
+          />
+        </q-td>
+      </q-tr>
+    </template>
```

Comment:
```suggestion
    <template #body-cell-title="p">
       <q-td :props="p">
          <router-link
            :to="{
              name: 'submission_review',
              params: { id: p.row.id },
            }"
            >{{ p.row.title }}
          </router-link>
        </q-td>
     </template>
     <template #body-cell-publication="p">
        <q-td :props="p">
          <router-link
            :to="{
              name: 'publication:home',
              params: { id: p.row.publication.id },
            }"
            >{{ p.row.publication.name }}
          </router-link>
        </q-td>
     </template>
     <template #body-cell-status="p">
        <q-td :props="p">
          {{ $t(`submission.status.${p.row.status}`) }}
        </q-td>
     </template>
     <template #body-cell-actions="p">
        <q-td :props="p">
          <submission-table-actions
            :submission="p.row"
            action-type="reviews"
            flat
          />
        </q-td>
     </template>
```
So, doing the overrides with #body-cell-[name] slots means that adding/rearranging a column won't break things and also means you don't have to output things that aren't getting transformed (e.g. the id column)

also, using using p for the slot scope variable means you don't have to rename props to propsVar in the setup

Reply:
I refactored this as suggested.